### PR TITLE
Added sync argument to Project constructor

### PR DESCRIPTION
**Pull request by shennan**
In OSX, the `ulimit` is hard-set by the OS, which maxes at 1024. This is the allowed number of open file descriptors at any one time. This can be reached fairly quickly on a project with many assets. Build scripts that use this library to copy many files, fail as a result. Adding a synchronous flag allows us to mitigate the issue in OSX platforms by passing a test for OSX as a third argument to Guilder:

`var project = Guilder(srcFolder, destFolder, /darwin/.test(process.platform))`

In a nutshell, this fixes copying issues in OSX by making the `runCopy` method synchronous when needed.

### CODE REVIEW

**Comment by binocarlos** [comment]
nice :)
